modules/nw-ingress-controller-nodeportservice-projects.adoc

```diff
@@ -0,0 +1,161 @@
+// Module included in the following assemblies:
+//
+// * networking/configuring_ingress_cluster_traffic/nw-configuring-ingress-controller-endpoint-publishing-strategy.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="nw-ingress-controller-nodeportservice-projects_{context}"]
+= Adding a single NodePort service to an Ingress Controller
+
+Instead of creating a `NodePort`-type `Service` for each project, you can create a custom Ingress Controller to use the `NodePortService` endpoint publishing strategy. To prevent port conflicts, consider this configuration for your Ingress Controller when you want to apply a set of routes, through Ingress sharding, to nodes that might already have a `HostNetwork` Ingress Controller.  
+
+Before you set a `NodePort`-type `Service` for each project, read the following considerations:
+
+* You must create a wildcard DNS record for the Ingress Controller. For more information about the required DNS records for routes, see "User-provisioned DNS requirements".
+* You must expose a route for your service and specify the `--hostname` argument for your custom Ingress Controller domain.
+* Ensure that all Ingress Controller routes can access your {product-title} cluster through the host IP address assigned to each node.
+* You must use the `curl` command to transfer data to the `NodePort`-type `Service` in the custom Ingress Controller.
+
+.Prerequisites
+
+* You installed the {oc-first}.
+* Logged in as a user with `cluster-admin` privileges.
+* You created a wildcard DNS record.
+// https://docs.openshift.com/container-platform/4.16/networking/ingress-controller-dnsmgt.html (does not detail how to create the DNS)
+
+.Procedure
+
+. Create a custom resource (CR) file for the Ingress Controller:
++
+.Example of a CR file that defines information for the `IngressController` object
+[source,yaml]
+----
+apiVersion: v1
+items:
+- apiVersion: operator.openshift.io/v1
+  kind: IngressController
+  metadata:
+    name: <custom_name> <1>
+    namespace: openshift-ingress-operator
+  spec:
+    replicas: 1
+    domain: <domain> <2>
+    nodePlacement:
+      nodeSelector:
+        matchLabels:
+          <key>: <value> <3>
+    namespaceSelector:
+     matchLabels:
+       <key>: <value> <3>
+    endpointPublishingStrategy:
+      type: NodePortService 
+kind: List
+metadata:
+  resourceVersion: ""
+  selfLink: ""
+# ...
+----
+<1> Specify the a custom `name` for the `IngressController` CR.
+<2> The DNS name that the Ingress Controller services.
+<3> Specify the label for a set of namespaces. Substitute `<key>:<value>` with a map of key-value pairs where `<key>` is a unique name for the new label and `<value>` is its value. For example: `ingresscontroller: custom-ic`.
+
+. Add a label to a node by using the `oc label node` command:
++
+[source,terminal]
+----
+$ oc label node <node_name> <key>=<ingress_controller_custom_resource> <1>
```

Comment:
```suggestion
$ oc label node <node_name> <key>=<value> <1>
```

Reply:
I think simply saying "value" is simpler and matches what we did for namespace labels below

Reply:
:heavy_check_mark: